metadata/metadata.go

```diff
@@ -105,3 +147,41 @@ func safeCopy(destination map[string]string, key, val string) {
 		destination[key] = val
 	}
 }
+
+// copyWithNewPrefix copies key/value pairs from a source map to a destination map where the key matches the specified prefix.
+// If replacementPrefix is different from prefix, the prefix will be replaced while performing the copy.
+func copyWithNewPrefix(src, dest map[string]string, prefix, replacementPrefix string) {
+	for key, value := range src {
+		if strings.HasPrefix(key, prefix) {
+			newKey := key
+			if prefix != replacementPrefix {
+				newKey = strings.Replace(key, prefix, replacementPrefix, 1)
+			}
+			dest[newKey] = value
+		}
+	}
+}
+
+// CopyLabelsByPrefix copies all labels from a source object to a destination object where the key matches the specified prefix.
+// If replacementPrefix is different from prefix, the prefix will be replaced while performing the copy.
+func CopyLabelsByPrefix(src, dest v1.Object, prefix, replacementPrefix string) {
```

Comment:
What about `CopyLabelsWithPrefix` for this one?

I propose these functions:
```
func CopyLabelsWithPrefix(source, destination v1.Object, sourcePrefix, destinationPrefix string)
func CopyLabelsWithPrefixReplacement(source, destination v1.Object, sourcePrefix, destinationPrefix string)
```

Reply:
I picked "CopyLabelsWithPrefixReplacement" as the new function name. Thanks